### ISSUE TITLE
Move unit tests to jest instead of cypress

### ISSUE THREAD

**Issue by HendrikThePendric** [dhis2/ui#91]
I think that adding e2e tests is very good and for a lot of things preferred over unit-tests. However, there are also some things that are more suited to unit testing. For example verifying that component props are mapped to element attributes correctly. This came up here https://github.com/dhis2/ui-core/pull/630#issuecomment-561148691.

This issue serves as a reminder for us to start adding unit tests where appropriate.

**Comment by ismay** [comment]
@HendrikThePendric I've changed the title slightly, hope you don't mind, I agree with your original statement, but think I'd word it more strongly even. I think we're finding now that the cypress tests cost a ton of time and maintenance. Initially our intention was to have human readable feature files, to ease interaction between design and dev.

We have the cucumber feature -> cypress test setup, so we can only have that setup with cypress. So we tended towards putting everything in cypress. I think that that's now proving to be quite expensive.

I would be in favour of moving as much tests as possible to jest. My proposal would be to do all our unit testing in jest, and then to have a a lot less, more realistic, high-level integration tests in cypress. In my opinion, since integration tests are so expensive, they should be the tip of the pyramid. Currently they represent the bulk of our test suite.
